forward_engineering/alterScript/alterScriptHelpers/columnHelpers/defaultValueHelper.js

```diff
@@ -0,0 +1,63 @@
+const { toPairs } = require('lodash');
+const { AlterScriptDto } = require('../../types/AlterScriptDto');
+const { getFullTableName, wrapInQuotes, wrapInSingleQuotes } = require('../../../utils/general');
+
+/**
+ * @param { ({ ddlProvider: Object, collection: Object }) }
+ * @returns { Array<AlterScriptDto> }
+ * */
+const getUpdatedDefaultColumnValueScriptDtos = ({ ddlProvider, collection }) =>
+	toPairs(collection.properties)
+		.filter(([_name, jsonSchema]) => {
+			const newDefault = jsonSchema.default;
+			const oldName = jsonSchema.compMod.oldField.name;
+			const oldDefault = collection.role.properties[oldName]?.default;
+			return newDefault && (!oldDefault || newDefault !== oldDefault);
```

Comment:
What if the `newDefault` for the numeric type is 0 ?

Reply:
I would also check a boolean column `false` default.

Reply:
Good point, updated and verified 👍🏻 

![image](https://github.com/user-attachments/assets/84bd9b49-cdd2-4be5-85a1-407b5b93d580)